bot/cogs/reddit.py

```diff
@@ -113,103 +120,63 @@ def __init__(self, bot: Bot):
-                new_posts = []
+    @loop()
+    async def top_daily_posts_loop(self) -> None:
+        """Post the top 5 posts daily."""
+        # once we upgrade to d.py 1.3 this can be removed and the loop can use the `time=datetime.time()` parameter
+        now = datetime.utcnow()
 
-                # Only show new posts if we've checked before.
-                if subreddit in self.last_ids:
-                    for post in posts:
-                        data = post["data"]
+        midnight_tomorrow = now.replace(day=now.day + 1, hour=0, minute=0, second=0)
+        seconds_until = (midnight_tomorrow - now).total_seconds()
 
-                        # Convert the ID to an integer for easy comparison.
-                        int_id = int(data["id"], 36)
+        await asyncio.sleep(seconds_until)
 
-                        # If we've already seen this post, finish checking
-                        if int_id <= self.last_ids[subreddit]:
-                            break
+        await self.bot.wait_until_ready()
+        if not self.webhook:
+            await self.bot.fetch_webhook(Webhooks.reddit)
 
-                        embed_data = {
-                            "title": textwrap.shorten(data["title"], width=64, placeholder="..."),
-                            "text": textwrap.shorten(data["selftext"], width=128, placeholder="..."),
-                            "url": self.URL + data["permalink"],
-                            "author": data["author"]
-                        }
+        for subreddit in RedditConfig.subreddits:
+            top_posts = await self.get_top_posts(subreddit=subreddit, time="day")
+            await self.webhook.send(username=f"{subreddit} Top Daily Posts", embed=top_posts)
 
-                        new_posts.append(embed_data)
-
-                self.last_ids[subreddit] = int(posts[0]["data"]["id"], 36)
-
-                # Send all of the new posts as spicy embeds
-                for data in new_posts:
-                    embed = Embed()
+    @loop()
+    async def top_weekly_posts_loop(self) -> None:
+        """Post a summary of the top posts every week."""
+        now = datetime.utcnow()
 
-                    embed.title = data["title"]
-                    embed.url = data["url"]
-                    embed.description = data["text"]
-                    embed.set_footer(text=f"Posted by u/{data['author']} in {subreddit}")
-                    embed.colour = Colour.blurple()
+        # Calculate the amount of seconds until midnight next monday.
+        # when we upgrade d.py this can be simplified to be run every midnight and just check against day
+        monday = now + timedelta(days=7 - now.weekday())
+        monday = monday.replace(hour=0, minute=0, second=0)
+        until_monday = (monday - now).total_seconds()
 
-                    await self.reddit_channel.send(embed=embed)
+        await asyncio.sleep(until_monday)
```

Comment:
I don't see a way to avoid some logic here by purely using the Task extension, as you only specify a target `time` not a day in the week. One way we could avoid having to calculate the delay and execute the sleep ourselves is by having the `daily` task check if it's Monday and, if so, trigger the weekly top posts function.

This means that the daily posts coroutine wakes up at every midnight and if it's Monday, it will also trigger the weekly top posts coroutine to do its business.